keras_cv/bounding_box/utils.py

```diff
@@ -40,13 +72,19 @@ def clip_to_image(bounding_boxes, images, bounding_box_format):
         ],
         axis=-1,
     )
-
+    areas = _relative_area(
+        clipped_bounding_boxes, bounding_box_format="rel_xyxy", images=images
+    )
     clipped_bounding_boxes = bounding_box.convert_format(
         clipped_bounding_boxes,
         source="rel_xyxy",
         target=bounding_box_format,
         images=images,
     )
+
+    clipped_bounding_boxes = tf.where(
+        tf.expand_dims(areas > 0.0, axis=-1), clipped_bounding_boxes, -1.0
```

Comment:
Should this be a specifiable threshold somehow? e.g. if you randomCrop an image and end up with a bounding box that's 5% of its original noise, do we really want to keep it?

(Maybe the answer is that this really should be exactly 0, I just wanted to check)

Reply:
The goal is to avoid fully invalid ones with this utility for now.  As of now, the models break if a box is fully outside of the image.  it produces a "NaN" in loss computation because it is impossible for models to predict boxes fully outside of the image.   Lets go ahead and submit this fix so Mosaic and RRC work, then we can add a configurable threshold going forward

Reply:
Ack sgtm.